mini-projects/01-device-driver/laptop-interface/src/commander.rs

```diff
@@ -0,0 +1,152 @@
+use crate::libserial::serial::Serial;
+use anyhow::Result;
+use std::thread;
+
+use crate::libserial::types::Port;
+use crate::threads::BufferedBackgroundSerial;
+
+#[allow(dead_code)]
+pub struct Commander {
+    serial: BufferedBackgroundSerial,
+    pos: (i32, i32),
+}
+
+unsafe impl Send for Commander {}
+
+#[allow(dead_code)]
+impl Commander {
```

Comment:
overall, I love this interface - make the api to command each device via serial very clean